src/main/java/de/doubleslash/keeptime/view/ProjectsListViewController.java

```diff
@@ -0,0 +1,395 @@
+      this.model = model;
+      this.controller = controller;
+      this.hideable = hideable;
+      this.mainStage = mainStage;
+      this.availableProjectsListView = availableProjectsListView;
+      availableProjectsListView.setCellFactory(listView -> returnListCellOfProject());
+
+      filteredData = new FilteredList<>(model.getSortedAvailableProjects(), p -> true);
+      availableProjectsListView.setItems(filteredData);
+
+      projectSelectionNodeMap = new HashMap<>(model.getAvailableProjects().size());
+
+      for (final Project project : model.getSortedAvailableProjects()) {
+         addProjectToProjectSelectionNodeMap(project);
+      }
+
+      searchTextField.textProperty().addListener((a, b, newValue) -> {
+         LOG.info("New filter value: " + newValue);
+         // TODO do i always have to create a new predicate?
+         filteredData.setPredicate(project -> {
+            // If filter text is empty, display all data.
+            if (newValue == null || newValue.isEmpty()) {
+               return true;
+            }
+
+            final String lowerCaseFilter = newValue.toLowerCase();
+
+            if (project.getName().toLowerCase().contains(lowerCaseFilter)) {
+               return true;
+            }
+
+            return false;
+         });
+         LOG.debug("Amount of projects to show '{}'.", filteredData.size());
+         availableProjectsListView.getSelectionModel().selectFirst();
+         availableProjectsListView.scrollTo(0);
+      });
+
+      searchTextField.setOnKeyPressed(eh -> {
+         final MultipleSelectionModel<Project> selectionModel = availableProjectsListView.getSelectionModel();
+         switch (eh.getCode()) {
+         case UP:
+            LOG.debug("Arrow up pressed.");
+            selectionModel.selectPrevious();
+            eh.consume();
+            break;
+         case DOWN:
+            LOG.debug("Arrow down pressed.");
+            selectionModel.selectNext();
+            eh.consume();
+            break;
+         case ESCAPE:
+            LOG.debug("Esc pressed.");
+            if (hideable) {
+               mainStage.hide();
+            }
+            eh.consume();
+            break;
+         default:
+            break;
+         }
+         availableProjectsListView.scrollTo(selectionModel.getSelectedIndex());
+      });
+
+      // enter pressed in textfield
+      searchTextField.setOnAction(ev -> {
+         final Project selectedItem = availableProjectsListView.getSelectionModel().getSelectedItem();
+         if (selectedItem != null) {
+            changeProject(selectedItem, 0);
+            searchTextField.setText("");
+         }
+      });
+
+      availableProjectsListView.getSelectionModel().setSelectionMode(SelectionMode.SINGLE);
+      availableProjectsListView.getSelectionModel().selectFirst();
+   }
+
+   private void addProjectToProjectSelectionNodeMap(final Project project) {
+      if (project.isEnabled()) {
```

Comment:
remove this condition, as this may result in unexpected behaviour (if you add something and it is not added). The method also does something, it does not tell you (only adding if its enabled) :)